components/director/internal/repo/query_builder.go

```diff
@@ -0,0 +1,111 @@
+package repo
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/jmoiron/sqlx"
+
+	"github.com/pkg/errors"
+)
+
+func getAllArgs(conditions Conditions) []interface{} {
+	var allArgs []interface{}
+
+	for _, cond := range conditions {
+		if argVal, ok := cond.GetQueryArgs(); ok {
+			allArgs = append(allArgs, argVal...)
+		}
+	}
+	return allArgs
+}
+
+func writeEnumeratedConditions(builder *strings.Builder, conditions Conditions) error {
+	if builder == nil {
+		return errors.New("builder cannot be nil")
+	}
+
+	var conditionsToJoin []string
+	for _, cond := range conditions {
+		conditionsToJoin = append(conditionsToJoin, cond.GetQueryPart())
+	}
+
+	builder.WriteString(fmt.Sprintf(" %s", strings.Join(conditionsToJoin, " AND ")))
+
+	return nil
+}
+
+// TODO: Refactor builder
+func buildSelectQuery(tableName string, selectedColumns string, conditions Conditions, orderByParams OrderByParams) (string, []interface{}, error) {
+	var stmtBuilder strings.Builder
+
+	stmtBuilder.WriteString(fmt.Sprintf("SELECT %s FROM %s", selectedColumns, tableName))
+	if len(conditions) > 0 {
+		stmtBuilder.WriteString(" WHERE")
+	}
+
+	err := writeEnumeratedConditions(&stmtBuilder, conditions)
+	if err != nil {
+		return "", nil, errors.Wrap(err, "while writing enumerated conditions")
+	}
+
+	err = writeOrderByPart(&stmtBuilder, orderByParams)
+	if err != nil {
+		return "", nil, errors.Wrap(err, "while writing order by part")
+	}
+
+	allArgs := getAllArgs(conditions)
+
+	return getQueryFromBuilder(stmtBuilder), allArgs, nil
+}
+
+const anyKeyExistsOp = "?|"
+const anyKeyExistsOpPlaceholder = "{{anyKeyExistsOp}}"
+
+const allKeysExistOp = "?&"
+const allKeysExistOpPlaceholder = "{{allKeysExistOp}}"
+
+const singleKeyExistsOp = "] ? "
+const singleKeyExistsOpPlaceholder = "{{singleKeyExistsOp}}"
+
+var tempReplace = []string{
+	anyKeyExistsOp, anyKeyExistsOpPlaceholder,
+	allKeysExistOp, allKeysExistOpPlaceholder,
+	singleKeyExistsOp, singleKeyExistsOpPlaceholder,
+}
+
+var reverseTempReplace = []string{
+	anyKeyExistsOpPlaceholder, anyKeyExistsOp,
+	allKeysExistOpPlaceholder, allKeysExistOp,
+	singleKeyExistsOpPlaceholder, singleKeyExistsOp,
+}
+
+// sqlx doesn't detect ?| and ?& operators properly
```

Comment:
I needed to do a temporary replace, because sqlx.Rebind doesn't detect operators.

So we could end up with this:
```
SELECT id, tenant_id, name, description, status_condition, status_timestamp, creation_timestamp FROM public.runtimes WHERE tenant_id = $1 AND id IN (SELECT "runtime_id" FROM public.labels WHERE "runtime_id" IS NOT NULL AND "tenant_id" = $2 AND "key" = $3 AND "value" **$4|** array[$5]) ORDER BY creation_timestamp ASC
```

The only solution for that would be switching to `db.NamedExec` and use named parameters (switch from `[]interface{}` in helpers to `map[string]interface{}`, but it would be too time consuming to do it - to generate/ensure param keys are unique etc. We can think about it in future.